File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_02_二叉树属性/LeetCode_0404.java
//404. 左叶子之和
package com.kurumi.binarytree._02_二叉树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0404 {

}
class Solution {
  public int sumOfLeftLeaves(TreeNode root) {
    if(root==null) return 0;
    int leftSum=0;
    if(root.left!=null&&root.left.left==null&&root.left.right==null) leftSum=root.left.val;
    return leftSum+sumOfLeftLeaves(root.left)+sumOfLeftLeaves(root.right);
  }
}

File: 卡码/代码随想录/1/06-二叉树/Go/02-二叉树属性/LeetCode_0404.go
//404. 左叶子之和
package main

import (
	. "binaryTree/types"
)

func sumOfLeftLeaves(root *TreeNode) int {
	if root==nil {
		return 0
	}
	leftSum:=0
	if root.Left!=nil&&root.Left.Left==nil&&root.Left.Right==nil {
		leftSum=root.Left.Val
	}
	return leftSum+sumOfLeftLeaves(root.Left)+sumOfLeftLeaves(root.Right)
}
